prqlc/prql-compiler/src/sql/dialect.rs

```diff
@@ -322,6 +561,27 @@ mod tests {
         )
         "###);
     }
+
+    #[rstest]
+    #[case("%Y-%m-%d", MsSqlDialect {}, "yyyy-MM-dd")]
+    #[case("%Y-%m-%d", MySqlDialect {},"%Y-%m-%d")]
+    #[case("%Y-%m-%d", PostgresDialect {}, "YYYY-MM-DD")]
+    #[case("%Y-%m-%d %H:%M:%S.%f", MsSqlDialect {}, "yyyy-MM-dd HH:mm:ss.ffffff")]
+    #[case("%Y-%m-%d %H:%M:%S.%f", MySqlDialect {}, "%Y-%m-%d %H:%i:%S.%f")]
+    #[case("%Y-%m-%d %H:%M:%S.%f", PostgresDialect {}, "YYYY-MM-DD HH24:MI:SS.US")]
+    // #[case("%Y-%m-%d %H:%M:%S.%f %z %Z", MsSqlDialect {}, "yyyy-MM-dd HH:mm:ss.ffffff")]
+    // #[case("%Y-%m-%d %H:%M:%S.%f %z %Z", MySqlDialect {}, "%Y-%m-%d %H:%i:%s.%f")]
+    // #[case("%Y-%m-%d %H:%M:%S.%f %z %Z", PostgresDialect {}, "YYYY-MM-DD HH24:MI:SS.US TZ TZD")]
+    fn test_translate_date(
+        #[case] prql_date_format: &str,
+        #[case] dialect: impl DialectHandler + 'static,
+        #[case] postgres_date_format: &str,
+    ) {
+        assert_eq!(
+            dialect.translate_prql_format(prql_date_format),
+            postgres_date_format.to_string()
+        );
+    }
```

Comment:
I'd prefer a much tests as possible in the integration test instead of unit test.

Reply:
(I agree on having integration tests, though I also find that initially having them as very small unit tests makes them easier to iterate on, though possibly that's a difference in programming styles — I'm more of the "write something wrong and change it until it works" approach!)

Reply:
Ofc @aljazerzen it's the plan (see the description)
I already made an integration test but just went with some unit tests as well to be faster and make sure the approach would be validated